### DISCUSSION TITLE
Versioning of pnpm packages

### DISCUSSION THREAD

**Discussion by zkochan** [pnpm/pnpm#3837] [Ideas]
The packages in the pnpm monorepo are independently versioned. As a result, it is hard to identify which versions are used by which pnpm CLI. For instance, `@pnpm/headless` is v16 but the current pnpm version is v6

We could version all packages with the major version of the main package (pnpm) multiplied by 100. So `@pnpm/headless` could be of version 616 but when pnpm v7 comes out, it will automatically bumped to v600. If `@pnpm/headless` will need breaking changes, we may release v617 in case pnpm is still of v6.

@GiladShoham this idea came to me when I was looking at the pnpm package versions in the Bit repo

**Comment by GiladShoham** [comment]
It will be great to have an SDK package, which will import all other packages, and expose the programmatic API from one organized place.
It will also prevent cases where you use 2 APIs from 2 different packages with different versions which are not compatible with each other.
That way you can also version this SDK packages with a version that follows the major version of the pnpm cli.

**Comment by zkochan** [comment]
So we have actually implemented this for pnpm v9 and pnpm v10. I totally forgot about this discussion and came up with the same idea again.
